contracts/HATVaultsRegistry.sol

```diff
@@ -0,0 +1,563 @@
+    * swapped to HAT tokens and sent to the hacker via a vesting contract
+    * @param _governanceHatReward The amount of the vault's native token to be
+    * swapped to HAT tokens and sent to governance
+    */
+    function addTokensToSwap(
+        IERC20 _asset,
+        address _hacker,
+        uint256 _hackersHatReward,
+        uint256 _governanceHatReward
+    ) external {
+        uint256 amount = _hackersHatReward + _governanceHatReward;
+        hackersHatReward[address(_asset)][_hacker] += _hackersHatReward;
+        governanceHatReward[address(_asset)] += _governanceHatReward;
+        _asset.safeTransferFrom(msg.sender, address(this), amount);
+    }
+
+    /**
+    * @notice Called by governance to swap vault's native tokens to HAT tokens
+    * and distribute the HAT tokens: Send to governance their share and send to
+    * beneficiaries their share through a vesting contract.
+    * @param _asset The vault's native token
+    * @param _beneficiaries Addresses of beneficiaries
+    * @param _amountOutMinimum Minimum amount of HAT tokens at swap
+    * @param _routingContract Routing contract to call for the swap
+    * @param _routingPayload Payload to send to the _routingContract for the
+    * swap
+    */
+    function swapAndSend(
+        address _asset,
+        address[] calldata _beneficiaries,
+        uint256 _amountOutMinimum,
+        address _routingContract,
+        bytes calldata _routingPayload
+    ) external onlyOwner {
+        // Needed to avoid a stack too deep error
+        SwapData memory swapData;
+        for (uint256 i = 0; i < _beneficiaries.length; i++) {
+            swapData.totalHackersHatReward += hackersHatReward[_asset][_beneficiaries[i]];
+        }
+        swapData.amount = swapData.totalHackersHatReward + governanceHatReward[_asset];
+        if (swapData.amount == 0) revert AmountToSwapIsZero();
+        IERC20 _HAT = HAT;
+        (swapData.hatsReceived, swapData.amountUnused) = swapTokenForHAT(IERC20(_asset), swapData.amount, _amountOutMinimum, _routingContract, _routingPayload);
+
+        governanceHatReward[_asset] = swapData.amountUnused * governanceHatReward[_asset] / swapData.amount;
+
+        for (uint256 i = 0; i < _beneficiaries.length; i++) {
+            uint256 hackerReward = swapData.hatsReceived * hackersHatReward[_asset][_beneficiaries[i]] / swapData.amount;
+            swapData.totalHackerReward += hackerReward;
+            hackersHatReward[_asset][_beneficiaries[i]] = swapData.amountUnused * hackersHatReward[_asset][_beneficiaries[i]] / swapData.amount;
+            address tokenLock;
+            if (hackerReward > 0) {
+                // hacker gets her reward via vesting contract
+                tokenLock = tokenLockFactory.createTokenLock(
+                    address(_HAT),
+                    0x000000000000000000000000000000000000dEaD, //this address as owner, so it can do nothing.
```

Comment:
wait we still ahve the tokenlock bug to resolve, set this to 0x0000?

Reply:
https://github.com/hats-finance/hats-contracts/pull/306